.github/workflows/CI.yml

```diff
@@ -506,11 +506,23 @@ jobs:
           prerelease: ${{ needs.setup_release.outputs.pre_release }}
 
   build_mac:
-    name: MacOS
-    runs-on: macos-11
     needs: [check_changelog, setup_release]
     env:
       BOOST_VERSION: 1.83.0
+    strategy:
+      fail-fast: false  # false to test all, true to fail entire job if any fail
+      matrix:
+        include:
+          # https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/about-github-hosted-runners#standard-github-hosted-runners-for-public-repositories
+          # while GitHub has larger macOS runners, they are not available for our repos :(
+          - os_version: "12"
+            arch: "x86_64"
+          - os_version: "13"
+            arch: "x86_64"
+          - os_version: "14"
+            arch: "arm64"
```

Comment:
`arch` field will be used in the generated dmg name... should we use `Intel` and `M1` instead?

Reply:
Apple Silicon is the official name, but I'm also fine with `arm64`. I don't think we should use `M1`. It used to be synonymous with Apple Silicon, but now with M2 and M3, I don't think it makes sense to continue using `M1` to mean Apple Silicon generally.